### PR TITLE
feat: embed common client requests into the template html

### PR DESCRIPTION
**Pull request by kylecarbs**
This should reduce the number of API requests a client makes when loading the dashboard dramatically!

### CODE REVIEW

**Comment by kylecarbs** [comment]
Force merging this to see it on dev.coder.com (I'm excited about it). If anyone finds anything post-merge, just comment and I'll fix!

**Comment by BrunoQuaresma** [comment]
Maybe one thing we can do to simplify is only have one tag called "state" or "initialData" that has all these info into a single JSON so we can have only one function that parses that and return the data instead of having them in multiple places.

File: site/src/xServices/appearance/appearanceXService.ts [hunk 1]
```diff
@@ -107,7 +107,20 @@ export const appearanceMachine = createMachine(
...
+        if (appearance) {
+          const rawContent = appearance.getAttribute("content")
+          try {
+            return JSON.parse(rawContent as string)
+          } catch (ex) {
+            // Ignore this and fetch as normal!
+          }
+        }
+
+        return API.getAppearance()
```
Comment (ammario) [inline site/src/xServices/appearance/appearanceXService.ts:122]: why fallback to API? when would the value not be injected in the HTML?

File: site/site.go [hunk 2]
```diff
@@ -246,28 +271,88 @@ func (h *handler) serveHTML(resp http.ResponseWriter, request *http.Request, req
...
+			user, err = h.opts.Database.GetUserByID(ctx, apiKey.UserID)
+			return err
+		})
+		eg.Go(func() error {
+			memberIDs, err := h.opts.Database.GetOrganizationIDsByMemberIDs(ctx, []uuid.UUID{apiKey.UserID})
+			if errors.Is(err, sql.ErrNoRows) || len(memberIDs) == 0 {
+				return nil
+			}
+			if err != nil {
+				return nil
```
Comment (ammario) [inline site/site.go:309]: If this is on purpose it deserves a comment

Comment (kylecarbs) [reply site/site.go:309]: Fair

**Pull request merged**